src/java/org/apache/cassandra/db/marshal/MapType.java

```diff
@@ -398,4 +375,37 @@ public ByteBuffer getMaskedValue()
     {
         return decompose(Collections.emptyMap());
     }
+
+    @Override
+    public List<ByteBuffer> filterSortAndValidateElements(List<ByteBuffer> buffers)
+    {
+        // We depend on Maps to be properly sorted by their keys, so use a sorted map implementation here.
+        SortedMap<ByteBuffer, ByteBuffer> map = new TreeMap<>(getKeysType());
+        Iterator<ByteBuffer> iter = buffers.iterator();
+        while (iter.hasNext())
+        {
+            ByteBuffer keyBytes = iter.next();
+            ByteBuffer valueBytes = iter.next();
+
+            if (keyBytes == null)
+                throw new MarshalException("null is not supported inside collections");
+
+            getKeysType().validate(keyBytes);
+
+            if (valueBytes == null)
+                throw new MarshalException("null is not supported inside collections");
+
+            getValuesType().validate(valueBytes);
+
```

Comment:
we will throw exception if at least one is null ,so what about doing check at the before ?

```
ByteBuffer keyBytes = iter.next();
  ByteBuffer valueBytes = iter.next();
  if (keyBytes == null || valueBytes == null)
                throw new MarshalException("null is not supported inside collections");
  getKeysType().validate(keyBytes);
  getValuesType().validate(valueBytes);

  map.put(keyBytes, valueBytes);
```

Reply:
Changed